tcms/bugs/static/bugs/js/mutable.js

```diff
@@ -35,18 +35,27 @@ $(document).ready(function () {
 function populateVersion () {
   const productId = $('#id_product').val()
 
-  const href = $('#add_id_version')[0].href
-  $('#add_id_version')[0].href = href.slice(0, href.indexOf('&product'))
-  $('#add_id_version')[0].href += `&product=${productId}`
-  $('#id_version').find('option').remove()
-  updateVersionSelectFromProduct()
+  if ($('#id_product').val() === null) {
+    $($('#add_id_version')[0]).addClass('disabled')
+  } else {
+    $($('#add_id_version')[0]).removeClass('disabled')
+    const href = $('#add_id_version')[0].href
+    $('#add_id_version')[0].href = href.slice(0, href.indexOf('&product'))
+    $('#add_id_version')[0].href += `&product=${productId}`
+    $('#id_version').find('option').remove()
```

Comment:
If the + button becomes disabled then user can't click on it so you don't really have to safeguard much of this with a conditional. IMO the only thing that needs the conditional is the add/remove of the disabled class.

Reply:
Note: if product isn't selected the URL should end in `&product=null` which shouldn't be a problem if you can't click on the link.